homeassistant/components/squeezebox_player/media_player.py

```diff
@@ -0,0 +1,223 @@
+
+
+class SqueezeboxPlayer(MediaPlayerEntity):
+    """Representation of MediaPlayerEntity from Squeezebox Player."""
+
+    _attr_should_poll = False
+    _attr_supported_features = SUPPORTED_FEATURES
+    _attr_device_class = MediaPlayerDeviceClass.SPEAKER
+
+    def __init__(self, slimserver: SlimServer, player: SlimClient) -> None:
+        """Initialize MediaPlayer entity."""
+        self.slimserver = slimserver
+        self.player = player
+        self._attr_unique_id = player.player_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.player.player_id)},
+            manufacturer=DEFAULT_NAME,
+            model=self.player.device_model or self.player.device_type,
+            name=self.player.name,
+            hw_version=self.player.firmware,
+        )
+        # PiCore player has web interface
+        if "-pCP" in self.player.firmware:
+            self._attr_device_info[
+                "configuration_url"
+            ] = f"http://{self.player.device_address}"
+        self.update_attributes()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        self.update_attributes()
+        self.async_on_remove(
+            self.slimserver.subscribe(
+                self._on_slim_event,
+                (
+                    EventType.PLAYER_UPDATED,
+                    EventType.PLAYER_CONNECTED,
+                    EventType.PLAYER_DISCONNECTED,
+                    EventType.PLAYER_NAME_RECEIVED,
+                    EventType.PLAYER_RPC_EVENT,
+                ),
+                player_filter=self.player.player_id,
+            )
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return availability of entity."""
+        return self.player.connected
+
+    @property
+    def state(self) -> str:
+        """Return current state."""
+        if not self.player.powered:
+            return STATE_OFF
+        return STATE_MAPPING[self.player.state]
+
+    @callback
+    def update_attributes(self) -> None:
+        """Handle player updates."""
+        self._attr_name = self.player.name
+        self._attr_volume_level = self.player.volume_level / 100
+        self._attr_media_position = self.player.elapsed_seconds
+        self._attr_media_position_updated_at = utcnow()
+        self._attr_media_content_id = self.player.current_url
+        self._attr_media_content_type = "music"
+
+    async def async_media_play(self) -> None:
+        """Send play command to device."""
+        await self.player.play()
+
+    async def async_media_pause(self) -> None:
+        """Send pause command to device."""
+        await self.player.pause()
+
+    async def async_media_stop(self) -> None:
+        """Send stop command to device."""
+        await self.player.stop()
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
```

Comment:
Round maybe?
```suggestion
        volume = round(volume * 100)
```

Reply:
> 

I understand what you're saying here but Logitech Media Server is one of the implementations of the slimproto (aka squeezebox) protocol and a standalone server. So actually the name of that integration should be "Logitech Media Server" and not squeezebox. For example "Roon" is also a server implementing slimproto and that is also a separate integration in HA.

What we did here is implementing the slim protocol ourselves in python as an alternative to Roon and Logitech media Server so the user doesn't need a media server at all, HA is the Media Server in this case.

I agree with you that it's a bit confusing for the user. My suggestion would be to rename the title of the current squeezebox integration to: **Squeezebox (Logitech Media Server)**